src/Shared/PackageManagers/BaseProjectManager.cs

```diff
@@ -127,7 +127,7 @@ public static async Task<JsonDocument> GetJsonCache(string uri, bool useCache =
                     }
                 }
             }
-
+            if (WebClient == null) { throw new NullReferenceException(nameof(WebClient)); }
```

Comment:
The point of nullable annotations is to make null reference accesses explicit. If I don't check here we get a warning on the next line because WebClient can be null before you try to GetAsync it so you need to handle it there anyway.

Reply:
Finally,
1. This is not a server. 
2. A case where we *are* throwing an exception, not silently eating it.  Its up to the consumer to handle it properly, but we can't do anything about WebClient being null.  If you want to throw this null reference exception in the constructor instead when you instantiate webclient and not allow it to be nullable I can make that change.  
3. I don't think this is a boneheaded exception or close to the example they're talking about.  We can't Environment.Exit here the closest thing we can do is throw an exception.  This is as "fail hard" as you can possibly get in this case, so we are already following their advice.

Using Nullable annotations you must handle every case of nulls to generally *avoid* NullReferenceException, though in this case we don't have any choice but to throw a NullReferenceException.

Reply:
I switched to handling this by throwing a NullReferenceException in the constructor instead.